### DISCUSSION TITLE
Comparing different months from same database

### DISCUSSION THREAD

**Discussion by PerryShafran-NOAA** [dtcenter/METplus#1011] [Plot Generation] [answered]
Hi, there,

A colleague wishes to be able to compare the same month of one year vs a different year (e.g. August 2019 vs August 2020) from the same database in METviewer.  What would be the easiest way to do this for a timeseries plot?  For a diurnal/dieoff plot?  

Thanks for any assistance anyone can provide.

Perry

**Comment by georgemccabe** [comment]
Hi Perry, thanks for posting your question here. I have asked @TatianaBurek to take a look at this question.

**Comment by TatianaBurek** [comment]
Perry
As I understand, you would like to compare different periods of time on the same plot.
Please take a look at job # 20210721_143436. I couldn't find the exact data so I improvised.
This plot compares periods of May 10-11 and May 14-15. I selected both periods as series variables and grouped them. 
Attached is the screen shot of the UI configuration:
![Screen Shot 2021-07-21 at 08 40 53](https://user-images.githubusercontent.com/21346768/126507861-348abdaa-c7ca-48ab-94b3-ee00d0b7148d.png)

**Reply by PerryShafran-NOAA** [reply]
Hi, Tatiana,

Is it possible to make a plot with the x-axis being, say, a time period (like April 1-30), but one trace being 2019 and the other trace being 2020?  This looks useful with forecast hour as the independent variable, but I'd like to know if I can have a time series  comparing the same month for different years.

Thanks!

Perry

**Comment by TatianaBurek** [comment, marked as answer]
I don't think that this is possible since METviewer doesn't support groupings for the independent variable (x-axis).
@JohnHalleyGotway Do you have any suggestions?

**Reply by PerryShafran-NOAA** [reply]
Is this possible if one year is in one database and a second year is in another database?  

Perry

**Reply by JohnHalleyGotway** [reply]
I agree, I do not think this is possible in the current version of METviewer. Right now, METviewer is already doing some post-processing of time info. The "VALID_HOUR" entry is derived from the "FCST_VALID_BEG" column in the MET output. One way we might handle this in the future is add new derived column(s) for the components of the valid time:
- VALID_YEAR... when FCST_VALID_BEG = 20210807_120000, define VALID_YEAR = 2021
- VALID_MDHMS... when FCST_VALID_BEG = 20210807_120000, define VALID_MDHMS =  0807_120000
(or whatever other name is preferred)

But that'd be new development that would need to be written up as a GitHub issue.
@TatianaBurek do you agree that'd enable Perry to make the plot he describes?

Or perhaps there's some other way to slice/dice the components of the timestamps without needing to store many variations of them?

**Reply by TatianaBurek** [reply]
For  VALID_HOUR METviewer doesn't add a special column but calculates this value using SQL function on the fly:
`HOUR(fcst_valid_beg)`
We could calculate VALID_YEAR as 
`YEAR(fcst_valid_beg)`
and VALID_MONTH as
`MONTH(fcst_valid_beg)`
It is harder with month_date_hour (VALID_MDHMS) field and might not be possible.
I think that VALID_YEAR and VALID_MONTH could help create the plot that  Perry is talking about with the limitation of the time period  being a month.

**Reply by PerryShafran-NOAA** [reply]
Thanks for all the info, I think we can mark this as answered.  This indeed would be great for a future version.  We often do comparisons from year-to-year or season-to-season and it would be useful to have this capability.

**Comment by TatianaBurek** [comment]
The GitHub issue  for METviewer was created:
https://github.com/dtcenter/METviewer/issues/312
